opensearch-operator/controllers/tlsController_test.go

```diff
@@ -0,0 +1,134 @@
+package controllers
```

Comment:
noticed that you do not create and delete the CRD on the start and end.
would love to get a small expiation about it :)

Reply:
The tests for the tls controller are not a fully-fledged integration test but more a unit test of that one controller. As such I don't need to create the OpenSearchCluster object in kubernetes., I just need to provide it as input to the controller reconcile function and verify what it does. Makes it easier and faster to test features in isolation.